exec/java-exec/src/main/java/org/apache/drill/exec/expr/fn/DrillFunctionRegistry.java

```diff
@@ -64,62 +76,134 @@
       .put("CONVERT_FROM", Pair.of(2, 2))
       .put("FLATTEN", Pair.of(1, 1)).build();
 
+  /** Registers all functions present in Drill classpath on start-up. All functions will be marked as built-in.*/
   public DrillFunctionRegistry(ScanResult classpathScan) {
+    validate(BUILT_IN, classpathScan);
+    register(BUILT_IN, classpathScan, this.getClass().getClassLoader());
+    if (logger.isTraceEnabled()) {
+      StringBuilder allFunctions = new StringBuilder();
+      for (DrillFuncHolder method: registryHolder.getAllFunctionsWithHolders().values()) {
+        allFunctions.append(method.toString()).append("\n");
+      }
+      logger.trace("Registered functions: [\n{}]", allFunctions);
+    }
+  }
+
+  /**
+   * Validates all functions, present in jars.
+   * Will throw {@link FunctionValidationException} if:
```

Comment:
This is a JavaDoc comment, so HTML. To display correctly, use:
<ol>
<li>Jar with same name...</li>
<li>...</li>
</ol>